datadayessentials/config/_config.py

```diff
@@ -1,233 +1,161 @@
-import pandas as pd
+import dataclasses
+from typing import Union
+from azure.appconfiguration import AzureAppConfigurationClient
 import os
-import yaml
-from pathlib import Path
-from ._base import ILocalConfig, IGlobalConfig
+from ._execution_environment_manager import ExecutionEnvironmentManager, ExecutionEnvironment
+from ._base import IAuthentication
+from ..utils import CoreCacheManager
+import json
 
 
-class GlobalConfig(IGlobalConfig):
-    """Class to manage the global configuration file. Which cannot be edited by the user, and is for storing any settings that any user should be able to access.
-    Example:
-        >>> global_config = GlobalConfig()
-        >>> global_config.read()
-    """
-
-    path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "global_config.yml")
-
-    def read(self):
-        global_config = self.load_from_path(self.path)
-        return global_config
-
-
-class LocalConfig(ILocalConfig):
-    """Class to manage the local configuration file. THe local configuration file contains any user specific azure resources, including resource groups and storage accounts.
-    Example:
-        >>> local_config = LocalConfig()
-        >>> local_config.read()
-    """
-
-    ENVIRONMENT = os.environ.get("AZURE_ENVIRONMENT_NAME", "dev")
-    DEFAULT_CONFIG = {"sync_with_remote": False}
+class AzureConfigAuthentication(IAuthentication):
+    def get_credentials(self):
```

Comment:
We might as well get rename this function get_azure_credentials and have it inherit the functionality from the super method. Rather than having two names where we dont need them here. Or even just make it a standalone function as it is only called once